frontend/composables/useURLParams.ts

```diff
@@ -25,18 +67,32 @@ export const useURLParams = () => {
     return val
   }
 
-  const setVal = (key: string, val: string | undefined) => {
+  const resolvePendingValues = () => {
+    const pvs = pendingValues.value
     const query = new URLSearchParams(stringifyQuery(router.currentRoute.value.query))
-    if (val) {
-      query.set(key, val)
-    } else {
-      query.delete(key)
+    for (const [key, val] of pvs) {
+      if (val) {
```

Comment:
I'd explicitly check for `val !== undefined` here, blank strings seem like a sharp edge here.

Reply:
Excellent catch.